dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,631 @@
+            return true;
+        } catch (DotDataException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private Object pushNewFile(final String groupName, final String path, final File file,
+            final Map<String, Object> extraMeta) {
+
+        try (final FileByteSplitter fileSplitter = new FileByteSplitter(file)) {
+
+            final HashBuilder objectHashBuilder = Encryptor.Hashing.sha256();
+            final List<String> chunkHashes = new ArrayList<>();
+
+            for (final Tuple2<byte[], Integer> bytesRead : fileSplitter) {
+
+                objectHashBuilder.append(bytesRead._1(), bytesRead._2());
+                final String chunkHash = Encryptor.Hashing.sha256().append
+                        (bytesRead._1(), bytesRead._2()).buildUnixHash();
+                chunkHashes.add(chunkHash);
+                new DotConnect().executeUpdate(this.getConnection(),
+                        "INSERT INTO storage_data(hash_id, data) VALUES (?, ?)",
+                        chunkHash,
+                        bytesRead._1().length == bytesRead._2() ?
+                                bytesRead._1() : this.chunkBytes(bytesRead._2(), bytesRead._1()));
+            }
+
+            final String objectHash = objectHashBuilder.buildUnixHash();
+            final StringWriter metaDataJsonWriter = new StringWriter();
+            this.objectMapper.writeValue(metaDataJsonWriter, extraMeta);
+            new DotConnect().executeUpdate(this.getConnection(),
+                    "INSERT INTO storage(hash, path, group_name, metadata) VALUES (?, ?, ?, ?)",
+                    objectHash, path, groupName, metaDataJsonWriter.toString());
+
+            int order = 1;
+            for (final String chunkHash : chunkHashes) {
+
+                new DotConnect().executeUpdate(this.getConnection(),
+                        "INSERT INTO storage_x_data(storage_hash, data_hash, data_order) VALUES (?, ?, ?)",
+                        objectHash, chunkHash, order++);
+            }
+
+            return true;
+        } catch (DotDataException | NoSuchAlgorithmException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private byte[] chunkBytes(final int bytesLength, final byte[] bytes) {
+
+        final byte[] chunkedArray = new byte[bytesLength];
+
+        System.arraycopy(bytes, 0, chunkedArray, 0, bytesLength);
+
+        return chunkedArray;
+    }
+
+    private byte[] fileToBytes(final File file) {
+
+        try (InputStream input = com.liferay.util.FileUtil
+                .createInputStream(file.toPath(), "none")) {
+
+            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            final byte[] buffer = new byte[1024];
+            for (int readNum; (readNum = input.read(buffer)) != -1; ) {
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)